### PR TITLE
chore: assign target for rust-cache

### PR DESCRIPTION
**Pull request by linonetwo**

### CODE REVIEW

**Comment by himself65** [comment]
Why are there always "3 cancelled" after pushing the new commits? Can we run them or ignore them because merge have to pass all of the checks

/cc @darkskygit 

image

**Comment by linonetwo** [comment]
Seems canceled due to high concurrency in our repo.

After this PR, build speed will increase, and the cancel rate will drop.

**Comment by himself65** [comment]
> Seems canceled due to high concurrency in our repo.
> 
> After this PR, build speed will increase, and the cancel rate will drop.

of course, canceling the previous run is needed. but it should cancel **current** runs

**Comment by himself65** [comment]
> Why are there always "3 cancelled" after pushing the new commits? Can we run them or ignore them because merge have to pass all of the checks
> 
> /cc @darkskygit
> 
> image

we have fixed this by differ the group-id

**Review by himself65 — APPROVED** [review]

File: .github/workflows/client-app.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ permissions:
   security-events: write
 
 concurrency:
-  group: release-ci-group
+  group: ${{ (github.event_name == 'pull_request' && github.event.pull_request.number) || github.head_ref }}
```
Comment (lawvs) [inline .github/workflows/client-app.yml:25]: Need `github.workflow` to distinguish from other CI

I suggest use
```suggestion
  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
```

**Comment by linonetwo** [comment]
Just finish my lunch...

**Comment by himself65** [comment]
> Just finish my lunch...

I have to merge this because it will boost up the CI

**Pull request merged**